### PR TITLE
chore(spec): add spec docs for config, alerts, and dashboards

### PR DESCRIPTION
**Pull request by jbeveland27**
Also fixed dashboard validation issues in all packs

### CODE REVIEW

File: docs/config-spec.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+full-description: string, optional
+
+# Tags for filtering / searching criteria
+tags: list(string), optional
+
+# Displayed in listings and in "my packs"
+thumbnail: string, optional
+
+# Only used if pack has a Nerdpack or a Dashboard
+screenshots: list(string), optional
```
Comment (aswanson-nr) [inline docs/config-spec.md:71]: Originally the screenshots just sat in the `dashboards` or `nerdpacks` directories, are we changing that model to be more explicit?

**Review by aswanson-nr — COMMENTED** [review]
Looks great overall! Can you add the character limits to fields like `description` and `short-description`? Since we haven't locked in image file types, I'm fine with not listing them here for now.

File: docs/config-spec.md (hunk 1 above)
Comment (jbeveland27) [reply docs/config-spec.md]: I removed `thumbnail` and `screenshots`. I think it makes more sense to have those just be files in the directory and not listed explicitly in the config file.

**Comment by jbeveland27** [comment]
> Looks great overall! Can you add the character limits to fields like `description` and `short-description`? Since we haven't locked in image file types, I'm fine with not listing them here for now.

@aswanson-nr I added limits in `utils/schema/main_config.json` (and filled out more of the main config schema)

**Comment by jbeveland27** [comment]
I split the dashboard fixes out into #95

File: docs/rulesets/config.yml [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+  ################
+  # Content
+  ################
+  title:
+    type: text
+    limit: 60
+  short-description:
+    type: text
+    limit: 120
```
Comment (jpvajda) [inline docs/rulesets/config.yml:43]:
```suggestion
    limit: 250
```

File: docs/config-spec.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+# uuid: guid, required
+
+# Name acts as a unique identifier for this observability pack
+# Formatted as hyphenated lowercase
+# Example: apache, golden-signals-for-web-servers
+name: string, required
+
+# Deprecate in favor of short-description?
+# A short form description for this Observability Pack. Used throughout the platform when displaying the pack.
+description: string, required
```
Comment (jpvajda) [inline docs/config-spec.md:31]: let's keep this field as description.

File: docs/config-spec.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+#####################
+# The name of the pack displayed every the pack is referenced
+title: string, optional
+
+# Displayed in search results and recommendations
+short-description: string, optional
+
+# Displayed on the pack overview/details page formatted into rich text.
+# Standardized to a template for the New Relic packs.
+full-description: string, optional
```
Comment (jpvajda) [inline docs/config-spec.md:65]:
```suggestion
description: string, optional
```

File: docs/config-spec.md [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+  - type: string, optional 
+    name: string, optional
+
+#####################
+# Content/Design
+#####################
+# The name of the pack displayed every the pack is referenced
+title: string, optional
+
+# Displayed in search results and recommendations
```
Comment (jpvajda) [inline docs/config-spec.md:60]:
```suggestion
# Displayed in search results and recommendations, summarizes a packs functionality
```

File: docs/config-spec.md [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+short-description: string, optional
+
+# Displayed on the pack overview/details page formatted into rich text.
+# Standardized to a template for the New Relic packs.
+full-description: string, optional
+
+# Tags for filtering / searching criteria
```
Comment (jpvajda) [inline docs/config-spec.md:67]:
```suggestion
# Tags for filtering / searching criteria (future release)
```

File: docs/config-spec.md [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+tags: list(string), optional
+
+# Displaying related child packs
+children: list(object), optional
+  - title: string, optional
+    description: string, optional
+    screenshots: string, optional
+
+# Contains avatar, GitHub username, and link to GitHub profile for each contributor
```
Comment (jpvajda) [inline docs/config-spec.md:76]:
```suggestion
# Contains avatar, GitHub username, and link to GitHub profile for each contributor (future release)
```

File: docs/config-spec.md [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+short-description: string, optional
+
+# Displayed on the pack overview/details page formatted into rich text.
+# Standardized to a template for the New Relic packs.
+full-description: string, optional
+
+# Tags for filtering / searching criteria
+tags: list(string), optional
+
+# Displaying related child packs
```
Comment (jpvajda) [inline docs/config-spec.md:70]:
```suggestion
# Displaying related child packs  (future release)
```

**Review by jpvajda — CHANGES_REQUESTED** [review]
made some suggestions based on our convo on this today.

**Pull request merged**